ObjectOrientedProgramming/Memory Ownership/move semantics/task.md

```diff
@@ -0,0 +1,54 @@
+Move semantics is a feature of C++ that allows you to efficiently transfer ownership of an object from one variable to another without copying the object. This can be useful for improving performance and avoiding unnecessary memory allocations.
+
+#### Move Semantics Overview
+- Efficient Resource Transfer. Move semantics allows the transfer of ownership from one object to another without the need for deep copying. This is especially beneficial for large or resource-intensive objects. 
+- Rvalue References. Move semantics utilize rvalue references (`&&`), which represent temporary objects or objects that are about to be destroyed. Rvalue references to facilitate the efficient transfer of ownership without unnecessary overhead.
```

Comment:
We have not introduced the concept of `lvalue` and `rvalue` in this course. 
This concept is quite complicated for newbies (especially if we also consider other newly added to C++ categories like `xvalue`). 

 So there are two solutions to this problem:
 - either try to explain `move` semantics without explicitly referring to `lvalues` and `rvalues`; 
 - properly introduce `lvalues` and `rvalues` here (before `move` semantics), and give more examples.